src/cli/subcommands/snapshot_cmd.rs

```diff
@@ -496,3 +516,59 @@ fn validation_spinner(prefix: &'static str) -> indicatif::ProgressBar {
     pb.enable_steady_tick(std::time::Duration::from_secs_f32(0.1));
     pb
 }
+
+async fn print_computed_state(
+    snapshot: PathBuf,
+    vm_height: ChainEpoch,
+    json: bool,
+) -> anyhow::Result<()> {
+    // Get header roots
+    let file = tokio::fs::File::open(&snapshot).await?;
+    let mut block_stream = CarStream::new(tokio::io::BufReader::new(file)).await?;
+    let roots = std::mem::take(&mut block_stream.header.roots);
+
+    // Initialize Blockstore
+    let store = Arc::new(AnyCar::try_from(snapshot)?);
+
+    // Prepare call to apply_block_messages
+    let ts = Tipset::load_required(&store, &TipsetKeys::new(roots.into()))?;
+
+    let genesis = ts.genesis(&store)?;
+    let network = if genesis.cid() == &*calibnet::GENESIS_CID {
```

Comment:
Can't help but wonder if we should actually introduce a global helper for this, I suspect this won't be the only place we would want to distinguish between the three.

Reply:
#3394